src/plugins/unified_histogram/public/chart/chart.tsx

```diff
@@ -267,6 +269,34 @@ export function Chart({
     });
   }
 
+  const removeTables = (attributes: LensAttributes) => {
+    const layers = attributes.state.datasourceStates.textBased?.layers;
```

Comment:
```suggestion
    if (!attributes.state.datasourceStates.textBased) {
       return attributes;
    }
    
    const layers = attributes.state.datasourceStates.textBased?.layers;
```

So it does not add `textBased` object for `formBased` case.